src/go/configgenerator/listener_generator.go

```diff
@@ -50,8 +50,87 @@ const (
 	statPrefix = "ingress_http"
 )
 
+// MakeListeners provides dynamic listeners for Envoy
+func MakeListeners(serviceInfo *sc.ServiceInfo) ([]*v2pb.Listener, error) {
+	var listeners []*v2pb.Listener
+
+	port := serviceInfo.Options.ListenerPort
+	isSsl := false
+	if serviceInfo.Options.SslPort != 0 {
+		port = serviceInfo.Options.SslPort
+		isSsl = true
+	}
+
+	listener, err := makeListener(serviceInfo, port, isSsl)
+	if err != nil {
+		return nil, err
+	}
+	listeners = append(listeners, listener)
+
+	redirect_listener, err := makeRedirectListener(serviceInfo)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Would be better to move the part after checking `serviceInfo.Options.SslPort != 0`

Reply:
Done